pkg/cloudprovider/aws/fake/ec2api.go

```diff
@@ -68,6 +77,9 @@ func (e *EC2API) CreateFleetWithContext(_ context.Context, input *ec2.CreateFlee
 	instances := []*ec2.Instance{}
 	instanceIds := []*string{}
 	for i := 0; i < int(*input.TargetCapacitySpecification.TotalTargetCapacity); i++ {
+		if e.ShouldTriggerInsufficientCapacity && aws.StringValue(input.LaunchTemplateConfigs[0].Overrides[0].InstanceType) == InsufficientCapacityInstanceType {
```

Comment:
Then this would be if `e.InsufficientCapacityExceptionInstanceTypes.Has(aws.StringValue(input.LaunchTemplateConfigs[0].Overrides[0].InstanceType))`